### ISSUE TITLE
About the normalization when evaluating the PCK accuracy

### ISSUE THREAD

**Issue by tjsmartlzh** [krrish94/CarKeypoints#12]
Hello!
I noticed that in you code you normalized the distance between predicted and target keypoints location by 1/10 of output resolution (64*64) when evaluating the PCK accuracy. 
However, I also noticed that in 'Viewpoints and Keypoints'(CVPR 2015) and 'Deep supervision with shape concepts for occlusion-aware 3d object parsing' (CVPR 2017), the keypoints prediction is considered to be correct 'if the corresponding prediction lies within α ∗ max(h, w) of the annotated keypoint with the corresponding object’s dimension being (h, w)'.
So could you please explain why you choose 1/10 of output resolution to normalize the distance?
Thank you very much!

**Comment by krrish94** [comment]
Could you please link the module / line in question?

**Comment by tjsmartlzh** [comment]
It is in valeval.lua, line 116: 
local dists = calcDists(preds, gt, torch.ones(preds:size(1))*opt.outputRes/10)
and in valeval.lua, line 41: 
dists[j][i] = torch.dist(label[i][j],preds[i][j])/normalize[i]
 It seems that you use 1/10 of output resolution to normalize the distance and use normalized distance to calculate PCK accuracy

**Comment by krrish94** [comment]
It's been >4 years, so I might be a little blurry on the details, but iirc, this normalization threshold was chosen to mimick the alpha PCK threshold (following the Caffe implementation of the "Viewpoints and Keypoints" paper). i.e., we would want points to be within 10% of the original dimensions.

A similar scheme was employed in the original stacked-hourglass (Lua) code, which we copied.

https://github.com/princeton-vl/pose-hg-demo/blob/master/main.lua

**Comment by tjsmartlzh** [comment]
Thank you very much!
So what kind of length metric did you use in your IROS (2018) paper (https://arxiv.org/pdf/1803.02057.pdf Page 7, column 2) to evaluate the PCK accuracy? 
Did you use 1/10 of output resolution or maximum of the object dimension?

**Comment by krrish94** [comment]
If I recall, we used a stricter criteria. Instead of using 0.1 * max(h, w), which is equal to 0.1 * 64, we used 0.1 * 64 * 64, which forces us to be with 10% error over the entire area of the image.

**Comment by tjsmartlzh** [comment]
That means the prediction is correct, when the distance between predicted location and target location  is lower than 0.1 * 64 * 64?

**Comment by krrish94** [comment]
Since the normalization term is in the denominator, it should be (1 / (0.1 * 64 * 64)), practically when they match to the pixel level :) (hence a very strict threshold)!

**Comment by tjsmartlzh** [comment]
But according to the definition of PCK accuracy, the criteria is: (distance between prediction and target)/(0.1 * 64 * 64) < threshold. And you use threshold = 0.1?

**Comment by krrish94** [comment]
The threshold is 0.5. https://github.com/krrish94/CarKeypoints/blob/master/valeval.lua#L87

Again, this was something we got from the Caffe impl, as far as I can recall. The only change we made is use 64 * 64, as opposed to 64

**Comment by tjsmartlzh** [comment]
I see. Thank you very much!
